functions/src/api/http/user.ts

```diff
@@ -0,0 +1,48 @@
+import { FastifyInstance } from "fastify/types/instance";
+import { decodeIdToken, isOrganizer } from "../../services/auth";
+import { addUserRole, removeUserRole } from "../../services/user";
+
+const basePath = "/user";
+
+const userRoutes = async (fastify: FastifyInstance) => {
+  fastify.patch(`${basePath}/:uid`, async (request, reply) => {
+    type Params = {
+      uid: string;
+    };
+    type Body = {
+      [key: string]: boolean;
+    };
+
+    const params = request.params as Params;
+    const body = request.body as Body;
+    if (!request.headers.authorization) {
+      return reply.unauthorized();
+    }
+    const callerUserUID = await decodeIdToken(request.headers.authorization);
+    if (!callerUserUID) {
+      return reply.unauthorized();
+    }
+    if (callerUserUID === params.uid) {
+      return reply.forbidden("You cannot update your own user data");
+    }
+    if (!(await isOrganizer(callerUserUID))) {
+      return reply.unauthorized();
+    }
+
+    if (body["organizer"]) {
```

Comment:
I would add all the possible keys in an array, and use it to check if we should add or remove the role. This way we can add any new role and we have one only reference to the string value (now we have 2)